core/src/test/java/org/apache/iceberg/io/TestByteBufferInputStreams.java

```diff
@@ -55,18 +56,22 @@ public void testReadAll() throws Exception {
     ByteBufferInputStream stream = newStream();
 
     int bytesRead = stream.read(bytes);
-    Assert.assertEquals("Should read the entire buffer", bytes.length, bytesRead);
+    Assertions.assertThat(bytesRead).as("Should read the entire buffer").isEqualTo(bytes.length);
 
     for (int i = 0; i < bytes.length; i += 1) {
-      Assert.assertEquals("Byte i should be i", i, bytes[i]);
-      Assert.assertEquals("Should advance position", 35, stream.getPos());
+      Assertions.assertThat(bytes[i]).as("Byte i should be i").isEqualTo((byte) i);
```

Comment:
This casting[(byte) i] is required. This is because in JUnit 4, the `assertEquals` method performs automatic type conversion when comparing values of different types. But in AssertJ, the `assertThat` method does not perform automatic type conversion.